app/services/idv/actions/verify_document_status_action.rb

```diff
@@ -0,0 +1,62 @@
+module Idv
+  module Actions
+    class VerifyDocumentStatusAction < Idv::Steps::VerifyBaseStep
+      def call
+        process_async_state(async_state)
+      end
+
+      private
+
+      def process_async_state(current_async_state)
+        render_json case current_async_state.status
+                    when :none
+                      { success: true, status: nil }
+                    when :in_progress
+                      { success: true, status: 'in_progress' }
+                    when :timed_out
+                      { success: false, status: nil, errors: ['timeout'] }
+                    when :done
+                      status = async_state_done(current_async_state)
+                      { success: true, status: status ? 'success' : 'fail' }
+                    end
+      end
+
+      def async_state_done(current_async_state)
+        result = current_async_state.result
+        return false unless process_result(result)
```

Comment:
WDYT of breaking this into separate methods? One predicate for "is it done?" and one that does all the side effects?